src/signTxOutput.c

```diff
@@ -468,10 +497,20 @@ static void signTxOutput_handleTokenAPDU(uint8_t* wireDataBuffer, size_t wireDat
 		read_view_t view = make_read_view(wireDataBuffer, wireDataBuffer + wireDataSize);
 
 		VALIDATE(view_remainingSize(&view) >= 4, ERR_INVALID_DATA);
-		token->assetNameSize = parse_u4be(&view);
+		const size_t newAssetNameSize = parse_u4be(&view);
+		PRINTF("  asst 1 >%.*H<\n", wireDataSize, wireDataBuffer);
+		PRINTF("  asst 2 >%d<\n", token->assetNameSize);
+		VALIDATE(ASSET_NAME_SIZE_MAX + 1 == token->assetNameSize || signTxOutput_lexiCompareViews(
+			make_read_view(token->assetNameBytes, token->assetNameBytes + token->assetNameSize),
+			make_read_view(wireDataBuffer + 4, wireDataBuffer + 4 + newAssetNameSize)
+			) < 0, ERR_INVALID_DATA);	//TODO should be STATE?
```

Comment:
`ERR_INVALID_STATE` is used for incorrect APDU, not just invalid data content of the APDU.

Perhaps you can replace the comment
`	// Request is not valid in the context of previous calls`
with
`	// Request is not valid in the context of previous calls (typically, wrong P1 or P2 in APDU msg)`